Runnect-iOS/Runnect-iOS/Network/Router/PublicCourseRouter.swift

```diff
@@ -1,8 +1,67 @@
 //
-//  PublicCourseRouter.swift
-//  Runnect-iOS
+//  PublicCourseRouter.swift
+//  Runnect-iOS
 //
-//  Created by sejin on 2023/02/16.
+//  Created by sejin on 2023/02/16.
 //
 
 import Foundation
+import Moya
+enum PublicCourseRouter {
+    case getCourseData
+    case getCourseSearchData(keyword: String)
+    case courseUploadingData(param: CourseUploadingRequestDto)
+    case getUploadedCourseDetail(publicCourseId: Int)
+    case getUploadedCourseInfo
+}
+
+extension PublicCourseRouter: TargetType {
+    var baseURL: URL {
+        guard let url = URL(string: Config.baseURL) else {fatalError("baseURL could not be configured")
+        }
+        return url
+        
+    }
+    var path: String {
+        switch self {
+        case .getCourseData, .courseUploadingData:
+            return "/public-course"
+        case .getCourseSearchData:
+            return "/public-course/search"
+        case .getUploadedCourseDetail(let publicCourseId):
+            return "/public-course/detail/\(publicCourseId)"
+        case .getUploadedCourseInfo:
+            return "/public-course/user"
+        }
+    }
+    var method: Moya.Method {
+        switch self {
+        case .getCourseData, .getCourseSearchData, .getUploadedCourseDetail, .getUploadedCourseInfo:
+            return .get
+        case .courseUploadingData:
+            return .post
+        }
+        
+    }
+    var task: Moya.Task {
+        switch  self {
+        case .getCourseSearchData(let keyword):
+            return .requestParameters(parameters: ["keyword": keyword], encoding: URLEncoding.default)
+        case .courseUploadingData(param: let param):
+            do {
+                return .requestParameters(parameters: try param.asParameter(), encoding: JSONEncoding.default)
+                
+            } catch {
+                fatalError("Encoding 실패")}
+        case .getCourseData, .getUploadedCourseDetail, .getUploadedCourseInfo:
+            return .requestPlain
+        }
+    }
+    var headers: [String: String]? {
+        switch self {
+        case .getCourseData, .courseUploadingData, .getUploadedCourseInfo, .getUploadedCourseDetail, .getCourseSearchData:
+            return Config.headerWithDeviceId
+            
+        }
```

Comment:
```suggestion
        switch self {
       default:
            return Config.headerWithDeviceId
        }
```
switch 문에서 모든 case가 같은 결과를 필요로 한다면 이렇게 default로 한번에 처리 할 수 있습니다!